### PR TITLE
[WIP] Started "v2", using composition model for config.

### PR DESCRIPTION
**Pull request by Doelfke**
Right now this is not destructive, but I assume we would want to change it to be, based on the suggestion to make the config composable, and the weight configurable.  I marked this as v2, with that assumption in mind.

This is POC really, but it should be easy enough to run with, if we want to take this approach.

I will leave comments below.

### CODE REVIEW

File: password-strength.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 'use strict';
 
+var module = {};
```
Comment (Doelfke) [inline password-strength.js]: I just added module for testing purposes, so I didn't have to use a build tool right now

File: password-strength.js [hunk 2]
```diff
@@ -131,8 +133,33 @@ var sequential = function sequential(sequence) {
...
+          fn: fn,
+          weight: weight
+        });
+
+        return config();
+      },
+      get: function () {
+        return scoreConfig;
+      },
+      checkPass: function (input) {
```
Comment (Doelfke) [inline password-strength.js]: Obviously checkPass doesn't belong here, but it highlights part of why v2 would have to be destructive.

File: password-strength.js [hunk 3]
```diff
@@ -131,8 +133,33 @@ var sequential = function sequential(sequence) {
...
 
+var config = function() {
+  var scoreConfig = [];
+
+  config = function () {
+    return {
+      add: function add (fn, weight) {
+        scoreConfig.push({
+          fn: fn,
+          weight: weight
```
Comment (Doelfke) [inline password-strength.js]: Not doing anything with weight yet

File: password-strength.js [hunk 4]
```diff
@@ -192,5 +228,8 @@ module.exports = {
     sequentialSymbols: sequentialSymbols,
 
     // wrapping it up
+    config: config,
     checkPassword: checkPassword
 };
+
+window.BdPasswordStrength = module.exports;
```
Comment (Doelfke) [inline password-strength.js]: Again, avoiding build tools ATM

File: password-strength.js [hunk 5]
```diff
@@ -131,26 +133,60 @@ var sequential = function sequential(sequence) {
...
-        consecutiveUppercase,
-        consecutiveLowercase,
-        consecutiveNumber,
-        sequentialLetters,
-        sequentialNumbers,
-        sequentialSymbols
-    ].reduce(function (totalScore, rule) {
-        return totalScore + rule(input);
+    var defaultConfig = [
+        { fn: numberOfCharacters, weight: 4 },
```
Comment (benmj) [inline password-strength.js]: I wonder whether a better compositional model is to have an object that contains the function and then the arguments array:

```
{ fn: numberOfCharacters, args: [4] }
```

This serves the purpose of giving the ability to customize the values of different functions -- which I think is a worthy goal -- but it leaves it open to having functions that take multiple parameters and don't just assume a single `weight` value. I also like how that would get the rule. weight out of the "runner" on line 189 of your PR.

**Pull request closed without merging**